### PR TITLE
feat(web builder): add test utils

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Description

add unwrap utils to stub builder

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by chenjiahan** [comment]
PR 的标题要改一下，会影响 squash 后的 commit message

File: packages/builder/web-builder/package.json [hunk 1]
```diff
@@ -45,7 +44,8 @@
     "terser-webpack-plugin": "5.3.3",
     "ts-loader": "^9.2.5",
     "tsconfig-paths-webpack-plugin": "^4.0.0",
-    "webpack": "^5.71.0"
+    "webpack": "^5.71.0",
+    "lodash": ">=4"
```
Comment (chenjiahan) [inline packages/builder/web-builder/package.json:48]: lodash 可以从 `@modern-js/utils/lodash` 里引用，不用单独安装

File: packages/builder/web-builder/src/core/createBuilder.ts [hunk 2]
```diff
@@ -23,11 +23,17 @@ export async function createBuilder(options?: BuilderOptions) {
...
+  const pluginStore = createPluginStore();
 
   await addDefaultPlugins(pluginStore);
 
+  let isFirstBuild = true;
   const build = async () => {
+    if (isFirstBuild) {
+      isFirstBuild = false;
+    } else {
+      throw new Error('Builders have been consumed.');
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/core/createBuilder.ts:35]: 假设前一次 build 已经结束了，也不能触发第二次 build 吗

File: packages/builder/web-builder/src/core/createContext.ts [hunk 3]
```diff
@@ -56,27 +58,43 @@ export async function createContext({
...
-      'rootPath',
-      'distPath',
-      'cachePath',
-      'configPath',
-      'tsconfigPath',
-      'originalConfig',
-      'framework',
-    ]),
-  );
+  const ctx = _.pick(context, [
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/core/createContext.ts:89]: 改成 lodash.pick 有啥优势吗

Comment (Asuka109) [reply packages/builder/web-builder/src/core/createContext.ts:89]: 刚才测试时候之前的 pick 一直返回空值，但是 lodash 的就没问题，待会再试试

File: packages/builder/web-builder/src/core/createBuilder.ts (hunk 2 above)
Comment (Asuka109) [reply packages/builder/web-builder/src/core/createBuilder.ts:35]: 我先回滚这里，另外开 mr 加个并发锁咋样

Comment (chenjiahan) [reply packages/builder/web-builder/src/core/createBuilder.ts]: 可以滴

File: packages/builder/web-builder/src/core/createContext.ts (hunk 3 above)
Comment (Asuka109) [reply packages/builder/web-builder/src/core/createContext.ts]: vitest mock 模块的时候会执行自动 mock 算法，把函数内部调用的 pick 也给 mock 成空函数了...

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**